File: leetcode-cn-1329/sort-the-matrix-diagonally.c
/*
 * sort-the-matrix-diagonally.c
 *
 *  Created on: Jan 31, 2020
 *      Author: yunfang
 */
#include "solution.h"

int main() {
	int r1[] = { 3, 3, 1, 1 };
	int r2[] = { 2, 2, 1, 2 };
	int r3[] = { 1, 1, 1, 2 };
	int *mat[] = { r1, r2, r3 };

	int m = 3;
	int n = 4;
	int returnSize = 0;
	int *returnColumnSizes = 0;
	diagonalSort(mat, m, &n, &returnSize, &returnColumnSizes);
}

File: leetcode-cn/src/com/leetcode/fangyun/p1329/SortTheMatrixDiagonally.java
package com.leetcode.fangyun.p1329;

public class SortTheMatrixDiagonally {

	public static void main(String[] args) {
		int[][] mat = { { 3, 3, 1, 1 }, { 2, 2, 1, 2 }, { 1, 1, 1, 2 } };
		print(mat);
		mat = new Solution().diagonalSort(mat);
		print(mat);
	}

	static void print(int[][] mat) {
		for (int i = 0; i < mat.length; i++) {
			for (int j = 0; j < mat[i].length; j++) {
				System.out.print(mat[i][j] + " ");
			}
			System.out.println();
		}
		System.out.println();
	}

}
